crates/evm/src/system_calls.rs

```diff
@@ -23,9 +24,113 @@ use revm_primitives::{
     ResultAndState, B256,
 };
 
+/// Apply the [EIP-2935](https://eips.ethereum.org/EIPS/eip-2935) pre block contract call.
+///
+/// This constructs a new [`Evm`] with the given database and environment ([`CfgEnvWithHandlerCfg`]
+/// and [`BlockEnv`]) to execute the pre block contract call.
+///
+/// This uses [`apply_blockhashes_contract_call`] to ultimately apply the blockhash contract state
+/// change.
+#[allow(clippy::too_many_arguments)]
+pub fn pre_block_blockhashes_contract_call<EvmConfig, DB>(
+    db: &mut DB,
+    evm_config: &EvmConfig,
+    chain_spec: &ChainSpec,
+    initialized_cfg: &CfgEnvWithHandlerCfg,
+    initialized_block_env: &BlockEnv,
+    block_number: u64,
+    block_timestamp: u64,
```

Comment:
I think we could obtain this from the blockenv?

Reply:
good catch, let me update the other syscalls to also do this